packages/material-ui/src/FieldTemplate/FieldTemplate.tsx

```diff
@@ -8,39 +8,67 @@ import List from "@material-ui/core/List";
 import ListItem from "@material-ui/core/ListItem";
 import Typography from "@material-ui/core/Typography";
 
+import WrapIfAdditional from "./WrapIfAdditional";
+
 const FieldTemplate = ({
   id,
   children,
+  classNames,
+  disabled,
   displayLabel,
+  hidden,
+  label,
+  onDropPropertyClick,
+  onKeyChange,
+  readonly,
   required,
   rawErrors = [],
   rawHelp,
   rawDescription,
+  schema,
 }: FieldTemplateProps) => {
+  if (hidden) {
+    return <div className="field-hidden">{children}</div>;
```

Comment:
This is indeed a needed change but I think it's beyond the scope of this PR, better to handle this in a separate issue as this fix is not 100% (for example the "add button" is still visible, existing props still take visual space instead of `disaply: none` etc).

Reply:
OK, sounds good.